vignettes/adrs_imwg.Rmd

```diff
@@ -0,0 +1,862 @@
+  "PD reason imaging", "any", "PD",
+  "PD reason serum/urine", "PD/death", "PD",
+  "PD reason serum/urine", "sCR/CR/VGPR/PR/MR/SD/NE/NA", "last confirmed status",
+)
+
+knitr::kable(list_resp)
+```
+
+`confirmed_response` function defined below takes as an argument data set with Overall Responses and returns data set with Confirmed Responses. 
+
+```{r}
+confirmed_response <- function(datain) {
+
+  data_adrs <- datain %>%
+    group_by(USUBJID) %>%
+    mutate(AVALC.next = lead(AVALC),
+           AVAL.next = lead(AVAL),
+           ADT.next = lead(ADT)) %>%
+    mutate(AVALC.confirmed = case_when(
+      # better response
+      AVALC %in% c("sCR", "CR", "VGPR", "PR", "MR") &
+        AVALC.next %in% c("sCR", "CR", "VGPR", "PR", "MR") &
+        (is.na(NACTDT) | ADT.next <= NACTDT) &
+        AVAL.next >= AVAL ~ AVALC,
+      # worse response
+      AVALC %in% c("sCR", "CR", "VGPR", "PR", "MR") &
+        AVALC.next %in% c("sCR", "CR", "VGPR", "PR", "MR", "SD") &
+        (is.na(NACTDT) | ADT.next <= NACTDT) &
+        AVAL.next < AVAL ~ AVALC.next,
+      # next assessment PD, NE, NA or after subsequent therapy
+      AVALC %in% c("sCR", "CR", "VGPR", "PR", "MR") &
+        (AVALC.next == "PD" | AVALC.next == "NE" | is.na(AVALC.next) |
+           !is.na(NACTDT) & ADT.next > NACTDT) ~ "SD",
+      # no need to confirm SD and NE
+      AVALC %in% c("SD", "NE") ~ AVALC,
+      # confirmed progression
+      AVALC == "PD" &
+        (PDIFL == "Y" | DTHPDFL == "Y" | PDOFL == "Y" & AVALC.next == "PD") ~
+        AVALC,
+      # unconfirmed progression
+      AVALC == "PD" & is.na(DTHPDFL) & PDOFL == "Y" &
+        (AVALC.next %in% c("sCR", "CR", "VGPR", "PR", "MR", "SD", "NE") |
+           is.na(AVALC.next)) ~ "NE")) %>%
+      mutate(AVAL.confirmed = aval_resp_conf(AVALC.confirmed)) %>%
+      # best Confirmed Response so far
+      mutate(AVAL.confirmed = cummax(AVAL.confirmed)) %>%
+      ungroup(USUBJID)
+
+  # char mapping to go back to AVALC values
+  avalc_resp_conf <- function(arg) {
+    case_match(
+      arg,
+      8 ~ "PD",
+      7 ~ "sCR",
+      6 ~ "CR",
+      5 ~ "VGPR",
+      4 ~ "PR",
+      3 ~ "MR",
+      2 ~ "SD",
+      1 ~ "NE",
+      NA_real_ ~ NA
+    )
+  }
+
+  data_adrs <- data_adrs %>%
+      mutate(AVALC.confirmed = avalc_resp_conf(AVAL.confirmed)) %>%
+      # if no Confirmed Response, assign SD
+      mutate(AVALC.confirmed = case_when(
+        AVALC.confirmed == "NE" ~ "SD",
+        TRUE ~ AVALC.confirmed
+      ))%>%
+      select(-AVAL, -AVALC, -AVAL.next, -AVALC.next, -ADT.next, 
+             -AVAL.confirmed) %>%
+      rename(AVALC = AVALC.confirmed) %>%
+      mutate(AVAL = aval_resp_imwg(AVALC)) %>%
+      mutate(
+        PARAMCD = "COVR",
+        PARAM = "Confirmed Response at Time Point by Investigator",
+        PARCAT1 = "Investigator",
+        PARCAT2 = "IMWG"
+      )
+}
+adrs_imwg <- confirmed_response(adrs) 
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
```

Comment:
701-1115 has PD on 1st record ( only labs related) and no subsequent assessment , so should be NE correct based on the workflow?

Reply:
PDIFL=Y